### DISCUSSION TITLE
How to make pnpm delete old versions of dependencies?

### DISCUSSION THREAD

**Discussion by behrangsa** [pnpm/pnpm#5808] [Q&A]
In my project, I have a dependency on a package named `resume-ts`:

```
"resume-ts": "^0.0.8",
```

On the file system, I have:

```
$ ls -alh node_modules/resume-ts
node_modules/resume-ts -> .pnpm/resume-ts@0.0.8/node_modules/resume-ts
```

In `.pnpm`, I have:

```
find node_modules/.pnpm/ -name resume-ts
node_modules/.pnpm/resume-ts@0.0.8/node_modules/resume-ts
```

If I bump up the package version:

```
"resume-ts": "^0.0.9",
```

and run `pnpm update`, I'll get:

```
$ find node_modules/.pnpm/ -name resume-ts
node_modules/.pnpm/resume-ts@0.0.9/node_modules/resume-ts
node_modules/.pnpm/resume-ts@0.0.8/node_modules/resume-ts
```

Is there a way to make pnpm delete the older version? I think VS Code has a bug that makes it keep on using the original sym link target, `node_modules/.pnpm/resume-ts@0.0.8/node_modules/resume-ts`, unless the window is reloaded.

**Comment by zkochan** [comment]
You may set [modules-cache-max-age](https://pnpm.io/npmrc#modules-cache-max-age) to `0`.

`pnpm prune` would also remove the old dep I believe.
